### PR TITLE
docs: init auth ms design

### LINKED ISSUE #2: auth microservice design
as a developer, I'd like a docs folder created that includes design docs for features in the current application

then, I'd like a design doc specifically constructed for the auth microservice. django already has auth, but we want to use a microservice to provide third party role information, which django maps to its own roles, similar to an SSO scenario

then, I'd like a docs/index.md which lists out all of the current designs in the repo

finally, I'd like the README updated to call out that /docs/index.md exists

### PR DESCRIPTION
**Pull request by Vandivier**
closes #2 

1. proposes a design of the auth microservice using Flask
2. creates `docs` folder with an index, template, and related updates to the README
3. fixes some incorrect info in the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,3 +28,22 @@ a full stack educational blog application written using three python web framewo
 6. related django tutorials:
    1. https://rasulkireev.com/managing-django-with-poetry/
    2. https://justdjango.com/blog/build-a-blog-with-django
+
+## contributing guidelines
+
+### before coding
+
+1. please contribute a design doc design doc for major features
+   1. a template can be found in `docs/template.md`
```
Comment (YuxuanMa-sys) [inline README.md:37]: Please add template file.

File: README.md [hunk 2]
```diff
@@ -21,3 +28,22 @@ a full stack educational blog application written using three python web framewo
...
+   2. existing designs are listed in `docs/index.md`
+2. TODO: define a major feature
```
Comment (YuxuanMa-sys) [inline README.md]: Please define a major feature.

File: README.md [hunk 3]
```diff
@@ -21,3 +28,22 @@ a full stack educational blog application written using three python web framewo
...
+3. TODO: define safe harbor for small changes where no design doc is needed
```
Comment (YuxuanMa-sys) [inline README.md]: Please define safe harbor fro small changes.

File: docs/2-auth-service-design.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+The preferred solution is to create a Flask endpoint that returns a rolename property with a value of "ADMIN" if the Django username includes the substring "admin". In all other cases, the third party authorization tool should return a value of "NONADMIN".
+
+Guest access is controlled on the Django side. The third party tool doesn't know about guest usage, and shouldn't even be consulted about guest permissions.
+
+### Alternatives Considered
+
+We also thought about using FastAPI instead of Flask.
+
+TODO: Why didn't we pick that?
```
Comment (YuxuanMa-sys) [inline docs/2-auth-service-design.md]: Please state why Flask is better in our case.

**Review by YuxuanMa-sys — COMMENTED** [review]
review done

File: README.md (hunk 2 above)
Comment (Vandivier) [reply README.md]: 1. alters db schema
2. if a change would result in more than 10% latency increase
3. any third party service integrations
4. any change involving more than 300 lines of code

**Review by YuxuanMa-sys — APPROVED** [review]

**Pull request merged**